src/mpi/datatype/looputil.c

```diff
@@ -365,12 +365,17 @@ int MPIR_Segment_init(const void *buf,
     return 0;
 }
 
-/* Segment_alloc
+/* MPIR_Segment_alloc
  *
  */
-struct MPIR_Segment *MPIR_Segment_alloc(void)
+struct MPIR_Segment *MPIR_Segment_alloc(const void *buf, MPI_Aint count, MPI_Datatype handle)
 {
-    return (struct MPIR_Segment *) MPL_malloc(sizeof(struct MPIR_Segment), MPL_MEM_DATATYPE);
+    struct MPIR_Segment *segp =
+        (struct MPIR_Segment *) MPL_malloc(sizeof(struct MPIR_Segment), MPL_MEM_DATATYPE);
+
+    segment_init(buf, count, handle, segp);
+
```

Comment:
Can we add a note here explaining that `segment_init ` always return 0, so we do not check the return value and handle the case where `segp` is allocated but `segment_init` fails ?

Reply:
Instead, I've changed the return type of `segment_init` to `void`.